src/Common/src/System/Net/Logging/GlobalLog.cs

```diff
@@ -0,0 +1,287 @@
+        {
+            EventSourceLogging.Log.FunctionStart(functionName, parameters);
+        }
+
+        public static void Assert(bool condition, string messageFormat, params object[] data)
+        {
+            if (!condition)
+            {
+                string fullMessage = string.Format(CultureInfo.InvariantCulture, messageFormat, data);
+                int pipeIndex = fullMessage.IndexOf('|');
+                if (pipeIndex == -1)
+                {
+                    Assert(fullMessage);
+                }
+                else
+                {
+                    int detailLength = fullMessage.Length - pipeIndex - 1;
+                    Assert(fullMessage.Substring(0, pipeIndex), detailLength > 0 ? fullMessage.Substring(pipeIndex + 1, detailLength) : null);
+                }
+            }
+        }
+
+        public static void Assert(string message)
+        {
+            Assert(message, null);
+        }
+
+        public static void Assert(string message, string detailMessage)
+        {
+            try
+            {
+                EventSourceLogging.Log.AssertFailed(message, detailMessage);
+            }
+            finally
+            {
+                Debug.Assert(false, message, detailMessage);
+            }
+        }
+
+        public static void Leave(string functionName)
+        {
+            EventSourceLogging.Log.FunctionStop(functionName);
+        }
+
+        public static void Leave(string functionName, string result)
+        {
+            EventSourceLogging.Log.FunctionStop(functionName, result);
+        }
+
+        public static void Leave(string functionName, int returnval)
+        {
+            EventSourceLogging.Log.FunctionStop(functionName, returnval.ToString());
+        }
+
+        public static void Leave(string functionName, bool returnval)
+        {
+            EventSourceLogging.Log.FunctionStop(functionName, returnval.ToString());
+        }
+
+        public static void Dump(byte[] buffer, int length)
+        {
+            Dump(buffer, 0, length);
+        }
+
+        public static void Dump(byte[] buffer, int offset, int length)
+        {
+            if (buffer == null)
+            {
+                EventSourceLogging.Log.WarningDumpArray("buffer is null");
+                return;
+            }
+            if (offset >= buffer.Length)
+            {
+                EventSourceLogging.Log.WarningDumpArray("offset out of range");
+                return;
+            }
+            if ((length < 0) || (length > buffer.Length - offset))
+            {
+                EventSourceLogging.Log.WarningDumpArray("length out of range");
+                return;
+            }
+            ArraySegment<byte> bufferSegment = new ArraySegment<byte>(buffer, offset, length);
+            byte[] bufferSegmentArray = bufferSegment.Array;
+            EventSourceLogging.Log.DebugDumpArray(bufferSegmentArray);
```

Comment:
This sequence is odd.  We're constructing an ```ArraySegment<byte>``` from a buffer, and then immediately accessing the buffer from that segment, and passing that buffer as the argument... the ArraySegment isn't used at all.  Maybe the author of these lines thought that ArraySegment would create a sub-array based on the offset and length and return that new instance from the Array property (it doesn't)?